Source/Core/AudioCommon/AudioCommon.cpp

```diff
@@ -35,9 +35,9 @@ namespace AudioCommon
 		std::string backend = SConfig::GetInstance().sBackend;
 		if (backend == BACKEND_OPENAL && OpenALStream::isValid())
 			g_sound_stream = new OpenALStream();
-		else if (backend == BACKEND_NULLSOUND && NullSound::isValid())
+		else if (backend.c_str() == BACKEND_NULLSOUND && NullSound::isValid())
```

Comment:
Actually, it does the complete opposite. Your new code checks to see if two pointers to c strings are identical, which they never will be because one is in a std::string.

The old code automatically converts a c string to a std::string and then calls the overloaded std::string equality operator, which compares the sting char by char, as you would expect.